index.bs

```diff
@@ -2725,6 +2769,284 @@ The [=remote end steps=] with <var ignore>session</var> and |command parameters|
 
 1. Return [=success=] with data null.
 
+</div>
+#### The browsingContext.locateNodes Command ####  {#command-browsingContext-locateNodes}
+
+The <dfn export for=commands>browsingContext.locateNodes</dfn> command returns a
+list of all nodes matching the specified locators.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browsingContext.LocateNodes = (
+        method: "browsingContext.locateNodes",
+        params: browsingContext.LocateNodesParameters
+      )
+
+      browsingContext.LocateNodesParameters = {
+         context: browsingContext.BrowsingContext,
+         locators: [ + browsingContext.Locator ],
+         ? startNode: script.SharedReference,
+         ? waitTimeout: js-uint
+      }
+      </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        browsingContext.LocateNodesResult = {
+            context: browsingContext.BrowsingContext,
+            nodes: [ * script.RemoteNodeValue ]
+        }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="locate nodes using CSS">
+To <dfn>locate nodes using CSS</dfn> with given |context|, |context nodes|,
+|selector| and <var ignore>session</var>:
+
+1. Let |returned nodes| be an empty array.
+
+1. For each |context node| of |context nodes|:
+
+  1. Let |elements| be the result of calling querySelectorAll() on |context| with |context node|
+     as this and |selector| as the argument. If this causes an exception to be thrown,
+     return [=error=] with [=error code=] [=invalid selector=]. 
+
+  1. Append |elements| to |returned nodes|
+
+1. Return |returned nodes|.
+
+</div>
+
+<div algorithm="locate nodes using XPath">
+To <dfn>locate nodes using XPath</dfn> with given |context|, |context nodes|
```

Comment:
should accept a document or realm instead of the context? 1) to avoid execution on a wrong document if it changes for the context 2) to allow locators to run a sandbox, so that the script locator can use the objects existing in the sandbox.

Reply:
Probably true. Please feel free to provide suggested spec prose for how to do that when all the user knows about is the contextId, since this is the `browserContext` module.

Reply:
I've changed the location algorithms to use `script.ContextTarget` instead of `browsingContext.BrowsingContext`. I think this might address this concern, along with the optional `sandbox` argument.

Reply:
@OrKoN These algorithms have been entirely rewritten. A re-review is required.